lib/commands/app-management.js

```diff
@@ -1,337 +1,172 @@
+  },
+  async activateApp(appId) {
+    const adb = /** @type {ADB} */ (this.adb);
+    this.log.debug(`Activating '${appId}'`);
+    const apiLevel = await adb.getApiLevel();
+    // Fallback to Monkey in older APIs
+    if (apiLevel < 24) {
+      // The monkey command could raise an issue as https://stackoverflow.com/questions/44860475/how-to-use-the-monkey-command-with-an-android-system-that-doesnt-have-physical
+      // but '--pct-syskeys 0' could cause another background process issue. https://github.com/appium/appium/issues/16941#issuecomment-1129837285
+      const cmd = ['monkey', '-p', appId, '-c', 'android.intent.category.LAUNCHER', '1'];
+      let output = '';
+      try {
+        output = await adb.shell(cmd);
+        this.log.debug(`Command stdout: ${output}`);
+      } catch (e) {
+        this.log.errorAndThrow(
+          `Cannot activate '${appId}'. Original error: ${/** @type {Error} */ (e).message}`
+        );
+      }
+      if (output.includes('monkey aborted')) {
+        this.log.errorAndThrow(`Cannot activate '${appId}'. Are you sure it is installed?`);
+      }
+      return;
     }
-  }
-  return APP_STATE.RUNNING_IN_BACKGROUND;
-};
 
-/**
- * @typedef {Object} MobileQueryAppStateOptions
- * @property {string} appId - Application package identifier. Must be always provided.
- */
+    let activityName = await adb.resolveLaunchableActivity(appId);
+    if (activityName === RESOLVER_ACTIVITY_NAME) {
+      // https://github.com/appium/appium/issues/17128
+      this.log.debug(
+        `The launchable activity name of '${appId}' was resolved to '${activityName}'. ` +
+          `Switching the resolver to not use cmd`
+      );
+      activityName = await adb.resolveLaunchableActivity(appId, {preferCmd: false});
+    }
 
-/**
- * Queries the current state of the app.
- *
- * @param {MobileQueryAppStateOptions} opts
- * @returns {number} Same as in `queryAppState`
- */
-commands.mobileQueryAppState = async function mobileQueryAppState (opts = {}) {
-  const { appId } = requireArgs('appId', opts);
-  return await this.queryAppState(appId);
-};
+    const stdout = await adb.shell([
+      'am',
+      apiLevel < 26 ? 'start' : 'start-activity',
+      '-a',
+      'android.intent.action.MAIN',
+      '-c',
+      'android.intent.category.LAUNCHER',
+      // FLAG_ACTIVITY_NEW_TASK | FLAG_ACTIVITY_RESET_TASK_IF_NEEDED
+      // https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_NEW_TASK
+      // https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_RESET_TASK_IF_NEEDED
+      '-f',
+      '0x10200000',
+      '-n',
+      activityName,
+    ]);
+    this.log.debug(stdout);
+    if (/^error:/im.test(stdout)) {
+      throw new Error(`Cannot activate '${appId}'. Original error: ${stdout}`);
+    }
+  },
+
+  async mobileActivateApp(opts) {
+    const {appId} = requireArgs('appId', opts);
+    return await this.activateApp(appId);
+  },
+  async removeApp(appId, opts = {}) {
```

Comment:
it is easier to read that if there is at least one line break between methods

Reply:
also, why the docstrings are now gone?

Reply:
all docstrings are moved into the interfaces.

Reply:
did a search/replace: ` {2}\},\n {2}(?=.*?\()` with `  },\n\n  `  😄